### DISCUSSION TITLE
Ideas

### DISCUSSION THREAD

**Discussion by B1ueAce** [Luke100000/minecraft-comes-alive#743] [Ideas]
This is my absolute favorite mod! I have played with it for several months now, so I have a good feel for useful potential features. I personally like the features specific to improving the village mechanics like making buildings purposeful and the impacts that the player's actions has on villagers.

**Buildings**

- Supply Ship/Wagon: Similar to the mine colonies mod, I think that adding some type of supply ship that costs emeralds to fund would be a useful structure to replenish common items like, stone, cobblestone, logs, wood, torches, or maybe food? I feel that this feature would challenge the player to charge more in taxes so that they can obtain more supply shipments to expand or improve the village. I want this feature the most out of all the others I will mention.

- Food Storage/Restaurant/Villager hunger: This feature could add a restaurant or food storage of some type that would feed villagers. This could be a new villager "need" that would challenge players to keep their village well fed by keeping the restaurant stocked, villagers would "take" a small portion of the food each day. If the food ever runs out, this could be a negative impact on the player's reputation in the village (decreasing hearts). *This feature should also support food from popular mods like farmer's delight and croptopia.

- School: The idea was first mentioned on the wiki discussion and I think this could be an improvement to children. The children in my Minecraft village meander around all day and keep me worried to death half of the time that they are going to get lost or killed by a mob lol. Adding a school would be a cool feature where children would safely congregate and could add a villager profession for "teacher".

- Merchant Stand/Shop: This would serve as a structure that a villager could sell more items from via their trade tab, these trades could be similar to what wandering traders offer. Some specific types of shops could be for armor, jewelers(instead of simply crafting a ring), weapons, or flower shops.

- Church: A building where clerics can work and where villagers can worship the ways of the Sirben. lol

**Professions**

- Teacher: as previously mentioned, this could be a profession for an adult villager who works in a school with the children.

- Merchant: This villager profession would be useful for better trades. These villagers would be stationed in their respective shops.

- Doctor: The cleric doesn't really fit with the infirmary in my opinion. I think that adding a doctor profession would be more appropriate for dealing with sick villagers. These villagers could sell items like potions of healing and golden apples. 

- Cook/ Waiter: This villager could work in the suggested restaurant building and sell food?

**Other Features**

- Villager chatter: It would be more immersive to walk in a village and hear them talking with one another, this could just be murmurs or even phrases that they repeat. It could be annoying but I think it would be cool. 

- Villager ranks: Having the ability as a king to rank up other villagers would be a minor feature that wouldn't really grant the other villagers any abilities but could be a useful feature for a player to "reward" loyal and long serving villagers. Ranks could simply follow the Roman Republic social classes (Plebian "lower" or Patricians "higher") or could be suited to their professions like (Chief Guard or Master Armorer). 

- Villager emigration: No matter how poorly treated a villager is they will always stay. Adding the possibility for villagers to leave the village because of harsh treatment, negative hearts, or as previously mentioned, failing to satisfy villager hunger, would challenge the player to care for their villagers so that they will stay.

Sorry for the overwhelming amount of suggestions, as I previously mentioned, I have just really enjoyed this mod for several months now and decided to make one big post with all of my ideas I have had while playing. I am more than happy to suggest more features as I see fit, or if all of these are all stupid I will just shut my mouth and continue to enjoy the mod in the direction the team wants it to go lol. Additionally, I tried looking for a good discord link, but all of the ones I have tried I haven't got to work. If I could get an invite link sent to me directly or commented down below that would be much appreciated. 

Best Wishes!

**Comment by JoceyWosey** [comment]
Dang, these are great ideas!
